### PR TITLE
Generate extension-api .d.ts and .js files using tsc

### LINKED ISSUE #1262: Export `@k8slens/extensions` for writing extension unit tests
**What would you like to be added**:

Export modules of `@k8slens/extensions` (currently has only `.d.ts`) for development/testing purposes only.

**Why is this needed**:

To be able to write unit tests for extensions.

Currently test throws

```
Cannot find module '@k8slens/extensions' from 'page.tsx'

Require stack:
 page.tsx
 page.test.tsx

> 1 | import { LensRendererExtension, Component } from "@k8slens/extensions";
```

### PR DESCRIPTION
**Pull request by jakolehm**
- generate `.d.ts` & `.js` files using `tsc`
- remove rollup (not used anymore)
- fixes #1262

### CODE REVIEW

File: src/common/utils/debouncePromise.ts [hunk 1]
```diff
@@ -4,6 +4,6 @@ export function debouncePromise<T, F extends any[]>(func: (...args: F) => T | Pr
   let timer: NodeJS.Timeout;
   return (...params: any[]) => new Promise((resolve, reject) => {
     clearTimeout(timer);
-    timer = setTimeout(() => resolve(func.apply(this, params)), timeout);
+    timer = global.setTimeout(() => resolve(func.apply(this, params)), timeout);
```
Comment (ixrock) [inline src/common/utils/debouncePromise.ts:7]: Why `global.setTimeout`? Any difference?

**Review by ixrock — APPROVED** [review]
Cool! And how it actually does it work? Is it because we started to re-export everything to  main/renderer api bundles? Haven't checked it in work yet though. Usually `tsc` compiler could generate such single file declarations only for AMD-modules or something.

**Comment by ixrock** [comment]
Helm test seems to be failed?
✕ ensures helm repos (60001 ms)

File: src/common/utils/debouncePromise.ts (hunk 1 above)
Comment (jakolehm) [reply src/common/utils/debouncePromise.ts:7]: Typescript complained about this:
> src/common/utils/debouncePromise.ts:7:5 - error TS2322: Type 'number' is not assignable to type 'Timeout'.

Basically typescript thinks it runs in a browser (`lib.DOM` is enabled) unless we explicitly say to use nodejs function.

**Comment by jakolehm** [comment]
> Cool! And how it actually does it work? Is it because we started to re-export everything to main/renderer api bundles? Haven't checked it in work yet though. Usually `tsc` compiler could generate such single file declarations only for AMD-modules or something.

This does _not_ generate single file declaration but does that really matter? Package.json is configured like this:
```
  "main": "dist/src/extensions/extension-api.js",
  "types": "dist/src/extensions/extension-api.d.ts",
```
which essentially means we are only exposing extension-api through `import "@k8slens/extensions`. Non-obfuscated d.ts/js is much more easy to explore from extension developers point of view.

**Comment by jakolehm** [comment]
Let's not merge this yet, it seems that generated js files need to be tweaked.

**Comment by jakolehm** [comment]
@ixrock @chenhunghan PTAL (again)

**Comment by chenhunghan** [comment]
LGTM

**Review by chenhunghan — APPROVED** [review]

File: tsconfig.extensions.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  "extends": "./tsconfig.json",
+  "files": [
+    "src/extensions/extension-api.ts",
+  ],
+  "compilerOptions": {
+    "module": "CommonJS",
+    "sourceMap": false,
+    "declaration": true
+  }
+}
```
Comment (ixrock) [inline tsconfig.extensions.json:11]:
```suggestion
{
  "extends": "./tsconfig.json",
  "include": [
    "types",
    "src/extensions/extension-api.ts"
  ],
  "compilerOptions": {
    "module": "CommonJS",
    "sourceMap": false,
    "declaration": true,
    "declarationDir": "src/extensions/npm/extensions/dist",
    "emitDeclarationOnly": true
  }
}
```

**Review by ixrock — APPROVED** [review]
Awesome! Works much better for DX and probably faster compilation than `rollup`.
Added `emitDeclarationOnly": true` to `tsconfig.extensions.json` since we don't need to publish *.js files to NPM (all served from runtime).

File: tsconfig.extensions.json (hunk 2 above)
Comment (chenhunghan) [reply tsconfig.extensions.json:11]: I think @jakolehm aims to fix this issue https://github.com/lensapp/lens/issues/1262 so these `.js` are needed (for the purpose of writing tests only)

Comment (jakolehm) [reply tsconfig.extensions.json:11]: Yes, we need `.js` files for test runtime.

Comment (ixrock) [reply tsconfig.extensions.json:11]: Alright, got it! 👍

**Pull request merged**